### DISCUSSION TITLE
Building for linux on 3.0-dev branch

### DISCUSSION THREAD

**Discussion by WhoAteDaCake** [lc-soft/LCUI#258] [General]
Hello, I tried to run 
```bash
xmake 
xmake package
xmake -P test helloworld
xmake run -P test helloworld
```
But the driver doesn't init anything, on further look it doesn't seem to define `USE_LIBX11` flag, which is set in `lib/platform/xmake.lua`:
```lua
    if has_package("libx11") then
        set_configvar("USE_LIBX11", 1)
    end
```

Am I doing wrong something, is there a flag to set, which I'm missing?

**Comment by lc-soft** [comment]
`LCUI_VIDEO_DRIVER_X11` is not replaced by `USE_LIBX11`, maybe I didn't commit the changes before.

**Reply by WhoAteDaCake** [reply]
Sorry, what do you mean? In the file `lib/platform/src/linux/linux_display.c` line 46, it's using `USE_LIBX11`

**Reply by lc-soft** [reply]
> Sorry, what do you mean? In the file `lib/platform/src/linux/linux_display.c` line 46, it's using `USE_LIBX11`

@WhoAteDaCake 

- Some configurations in `lib/platform/xmake.lua` are incorrect.
- Some source files are not modified correctly.

I'll solve them in a few days.

**Reply by WhoAteDaCake** [reply]
Thanks for the clarification! I'll go back to using the primary branch in the meanwhile.

**Reply by lc-soft** [reply]
I updated the xmake.lua file and some source files, now you can try again:

1. Delete `include/config.h` and `include/LCUI/config.h` file, If these files exist.
2. Delete `build` directory.
3. run `xmake`
4. run `xmake run helloworld`

**Reply by WhoAteDaCake** [reply]
Amazing, that is so much easier than the previous build process ! Thank you so much for that :+1:
